src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -26,20 +26,6 @@ public static partial class JsonSerializer
             if ((state.Current.IsProcessingDictionary || state.Current.IsProcessingIDictionaryConstructible) &&
                 state.Current.JsonClassInfo.DataExtensionProperty != state.Current.JsonPropertyInfo)
             {
-                string keyName = reader.GetString();
-
-                if (options.DictionaryKeyPolicy != null)
-                {
-                    keyName = options.DictionaryKeyPolicy.ConvertName(keyName);
-
-                    if (keyName == null)
-                    {
-                        ThrowHelper.ThrowInvalidOperationException_SerializerDictionaryKeyNull(options.DictionaryKeyPolicy.GetType());
-                    }
-
-                    keyName = options.DictionaryKeyPolicy.ConvertName(keyName);
```

Comment:
Out of curiosity, why did we initially try to get the keyName by calling ConvertName twice?